torch/csrc/distributed/rpc/init.cpp

```diff
@@ -472,11 +472,13 @@ PyObject* rpc_init(PyObject* /* unused */) {
               std::string /* selfName */,
               worker_id_t /* selfId */,
               int /* worldSize */,
+              std::shared_ptr<::c10d::ProcessGroup> /* processGroup */,
```

Comment:
Is this API what we plan to release in v1.6?

Since we don't need this until shutdown, I wonder if we can create this process group in `join` internally instead of asking applications to pass this in. In the args, we already have `rank` and `world_size`, can we let `rank=0` figure out a port, and send that port to all other peers, then they all use that as the `init_method` to create the process group instance.

Reply:
Personally I didn't consider this to be part of the API. Final users are supposed to enable the TensorPipe agent by calling:
```
rpc.init_rpc(name="foo", rank=0, world_size=2, backend=rpc.backend_registry.BackendType.TENSORPIPE, rpc_backend_options=rpc.TensorPipeRpcBackendOptions(...))
```
So I consider the API to consist of:
- the "name" `TENSORPIPE`,
- the arguments of `TensorPipeRpcBackendOptions`.

The constructor of the TensorPipeAgent is only invoked by the `_tensorpipe_init_backend_handler` and I considered it to be sort-of a "private" function.

Does this make sense? Or is the constructor of the agent a public API?

Reply:
As for constructing the process group on-demand, we could definitely do that, I just thought it was easier to construct it in Python (by copying the code used to construct it for the ProcessGroup agent) and then pass it to the agent. I can do as you prefer, but if you want me to construct it in C++ you'll have to show me how to do so. :)

Reply:
I see. This is good enough then. Let's land as-is.